cmd/agent/subcommands/check/command.go

```diff
@@ -0,0 +1,591 @@
+	cmd.Flags().BoolVarP(&formatTable, "table", "", false, "format aggregator and check runner output as an ascii table")
+	cmd.Flags().StringVarP(&breakPoint, "breakpoint", "b", "", "set a breakpoint at a particular line number (Python checks only)")
+	cmd.Flags().BoolVarP(&profileMemory, "profile-memory", "m", false, "run the memory profiler (Python checks only)")
+	cmd.Flags().BoolVar(&fullSketches, "full-sketches", false, "output sketches with bins information")
+	cmd.Flags().BoolVarP(&saveFlare, "flare", "", false, "save check results to the log dir so it may be reported in a flare")
+	cmd.Flags().UintVarP(&discoveryTimeout, "discovery-timeout", "", 5, "max retry duration until Autodiscovery resolves the check template (in seconds)")
+	cmd.Flags().UintVarP(&discoveryRetryInterval, "discovery-retry-interval", "", 1, "(unused)")
+	cmd.Flags().UintVarP(&discoveryMinInstances, "discovery-min-instances", "", 1, "minimum number of config instances to be discovered before running the check(s)")
+	config.Datadog.BindPFlag("cmd.check.fullsketches", cmd.Flags().Lookup("full-sketches")) //nolint:errcheck
+
+	// Power user flags - mark as hidden
+	createHiddenStringFlag(cmd, &profileMemoryDir, "m-dir", "", "an existing directory in which to store memory profiling data, ignoring clean-up")
+	createHiddenStringFlag(cmd, &profileMemoryFrames, "m-frames", "", "the number of stack frames to consider")
+	createHiddenStringFlag(cmd, &profileMemoryGC, "m-gc", "", "whether or not to run the garbage collector to remove noise")
+	createHiddenStringFlag(cmd, &profileMemoryCombine, "m-combine", "", "whether or not to aggregate over all traceback frames")
+	createHiddenStringFlag(cmd, &profileMemorySort, "m-sort", "", "what to sort by between: lineno | filename | traceback")
+	createHiddenStringFlag(cmd, &profileMemoryLimit, "m-limit", "", "the maximum number of sorted results to show")
+	createHiddenStringFlag(cmd, &profileMemoryDiff, "m-diff", "", "how to order diff results between: absolute | positive")
+	createHiddenStringFlag(cmd, &profileMemoryFilters, "m-filters", "", "comma-separated list of file path glob patterns to filter by")
+	createHiddenStringFlag(cmd, &profileMemoryUnit, "m-unit", "", "the binary unit to represent memory usage (kib, mb, etc.). the default is dynamic")
+	createHiddenStringFlag(cmd, &profileMemoryVerbose, "m-verbose", "", "whether or not to include potentially noisy sources")
+	createHiddenBooleanFlag(cmd, &generateIntegrationTraces, "m-trace", false, "send the integration traces")
+
+	cmd.SetArgs([]string{"checkName"})
+}
+
+// Commands returns a slice of subcommands for the 'agent' command.
+func Commands(globalArgs *command.GlobalArgs) []*cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "check <check_name>",
+		Short: "Run the specified check",
+		Long:  `Use this to run a specific check with a specific rate`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Could we extract that function so we have the cobra/`Commands` logic and the check command internal separated.

It might be out of scope for this PR, feel free to ignore.

Reply:
That will happen in the next set of PRs:
```go
		RunE: func(cmd *cobra.Command, args []string) error {
			// handle the deprecated --log-level flag
			if cliParams.logLevel != "" {
				os.Setenv("DD_LOG_LEVEL", cliParams.logLevel)
			}
			cliParams.cmd = cmd
			cliParams.args = args
			return fxutil.OneShot(run,
				fx.Supply(cliParams),
				fx.Supply(core.BundleParams{
					ConfFilePath:      globalParams.ConfFilePath,
					ConfigLoadSecrets: true,
				}),
				core.Bundle,
			)
		},
```